src/game/object/area.cpp

```diff
@@ -147,11 +147,15 @@ void Area::loadARE(const GffStruct &are) {
 void Area::loadCameraStyle(const GffStruct &are) {
     int styleIdx = are.getInt("CameraStyle");
     shared_ptr<TwoDaTable> styleTable(Resources::instance().get2DA("camerastyle"));
+    _cameraStyle.load(styleTable->rows().at(styleIdx));
 
-    _cameraStyle.distance = styleTable->getFloat(styleIdx, "distance", 0.0f);
-    _cameraStyle.pitch = styleTable->getFloat(styleIdx, "pitch", 0.0f);
-    _cameraStyle.viewAngle = styleTable->getFloat(styleIdx, "viewangle", 0.0f);
-    _cameraStyle.height = styleTable->getFloat(styleIdx, "height", 0.0f);
+    auto combatStyleRow = styleTable->findRowByColumnValue("name", "Combat");
+    if (combatStyleRow) {
+        _combatCamStyle.load(*combatStyleRow);
+    }
+    else {
+        throw out_of_range("Combat camera style failed to load.");
```

Comment:
nit: "out_of_range" is arguably more applicable to indexed collections. I'd use a "logic_error" here instead.

Reply:
Also, log messages should be stylized as follows:

"Area: failed to load combat camera style"